### PR TITLE
Deprecate some unused/untested drivers and packages

### PR DESCRIPTION
**Pull request by hmenke**
**Motivation for this change**

- deprecated pgfsys-textures.def
- deprecated pgfsys-vtex.def
- deprecated pgfutil-common-lists.tex

**Checklist**

Please [signoff your commits][git-s] to explicitly state your agreement to the [Developer Certificate of Origin][DCO]. If that is not possible you may check the boxes below instead:

- [x] Code changes are licensed under [GPLv2][GPL] + [LPPLv1.3c][LPPL]
- [x] Documentation changes are licensed under [FDLv1.2][FDL]

[git-s]: https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s
[DCO]: https://developercertificate.org
[GPL]: https://www.gnu.org/licenses/gpl-2.0.html
[LPPL]: https://www.latex-project.org/lppl/lppl-1-3c.txt
[FDL]: https://www.gnu.org/licenses/fdl-1.2.html

### CODE REVIEW

File: doc/generic/pgf/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ lot of contributed changes. Thanks to everyone who volunteered their time!
   `\pgfrevision{,date}` and `\pgfversion{,date}` are synonyms for now, but the
   revision should eventually gain back its original meaning.  However, as of now
   this is not supported by l3build.
+- The `textures` and `vtex` drivers have been deprecated. Both engines and no
```
Comment (muzimuzhi) [inline doc/generic/pgf/CHANGELOG.md]:
```suggestion
- The `textures` and `vtex` drivers have been deprecated. Both engines are no
```

**Review by muzimuzhi — APPROVED** [review]
Then `pgfplots` may need to move [`pgfplotsoldpgfsupp_pgfutil-common-lists.tex`](https://github.com/pgf-tikz/pgfplots/blob/master/tex/generic/pgfplots/oldpgfcompatib/pgfplotsoldpgfsupp_pgfutil-common-lists.tex), which is functionally identical to the one in `pgf` here, out from `oldpgfcompatib` sub-directory.

**Comment by hmenke** [comment]
That has already sort of happened: https://github.com/pgf-tikz/pgfplots/commit/daf5aa88ea1d97cc00a10222032381ac41e2e5c2

I've also grepped the TL 2021 tree and found that pgfplots is the only client using `pgfapplistnewempty`, so this change should not affect anyone else.

**Comment by muzimuzhi** [comment]
ready to merge?

**Pull request merged**